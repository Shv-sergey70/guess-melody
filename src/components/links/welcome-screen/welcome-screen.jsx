import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {ActionCreator} from "../../../reducer/game/game";
import Simple from "../simple/simple";
import Route from '../../../routes';

const WelcomeScreen = ({onClick}) => {
  return (
    <Simple
      to={Route.MAIN}
      className={`game__back`}
      onClick={onClick} >
      <span className="visually-hidden">Сыграть ещё раз</span>
      <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию"/>
    </Simple>
  );
};

WelcomeScreen.propTypes = {
  onClick: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onClick: () => dispatch(ActionCreator.resetState())
});

export {WelcomeScreen};
export default connect(null, mapDispatchToProps)(WelcomeScreen);
