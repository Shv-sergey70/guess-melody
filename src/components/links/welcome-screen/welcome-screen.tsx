import * as React from 'react';
import {connect} from "react-redux";
import {ActionCreator as GameActionCreator} from "../../../reducer/game/game";
import {ActionCreator as UserAnswersActionCreator} from '../../../reducer/user-answers/user-answers';
import Simple from "../simple/simple";
import Route from '../../../routes';

type Props = {
  onClick: (evt: React.SyntheticEvent) => void
}

const WelcomeScreen: React.FunctionComponent<Props> = ({onClick}) => {
  return (
    <Simple
      to={Route.MAIN}
      className={`game__back`}
      onClick={onClick} >
      <span className="visually-hidden">Сыграть ещё раз</span>
      <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию"/>
    </Simple>
  );
};

const mapDispatchToProps = (dispatch) => ({
  onClick: () => {
    dispatch(GameActionCreator.resetState());
    dispatch(UserAnswersActionCreator.resetAnswers());
  }
});

export {WelcomeScreen};
export default connect(null, mapDispatchToProps)(WelcomeScreen);
