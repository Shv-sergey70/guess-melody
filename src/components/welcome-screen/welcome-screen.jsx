import React from 'react';
import PropTypes from 'prop-types';
import {ActionCreator} from "../../reducer/game/game";
import {connect} from "react-redux";
import {getTime} from "../../reducer/game/selectors";

const WelcomeScreen = (props) => {
  const {
    time,
    attempts,
    onWelcomeButtonClick
  } = props;

  return (
    <section className="welcome">
      <div className="welcome__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/>
      </div>
      <button className="welcome__button" onClick={onWelcomeButtonClick}><span className="visually-hidden">Начать игру</span></button>
      <h2 className="welcome__rules-title">Правила игры</h2>
      <p className="welcome__text">Правила просты:</p>
      <ul className="welcome__rules-list">
        <li>За {Math.floor(time / 60)} минут нужно ответить на все вопросы.</li>
        <li>Можно допустить {attempts} ошибки.</li>
      </ul>
      <p className="welcome__text">Удачи!</p>
    </section>
  );
};

WelcomeScreen.propTypes = {
  time: PropTypes.number.isRequired,
  attempts: PropTypes.number.isRequired,
  onWelcomeButtonClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  time: getTime(state)
});

const mapDispatchToProps = (dispatch) => ({
  onWelcomeButtonClick: () => dispatch(ActionCreator.incrementStep())
});

export {WelcomeScreen};
export default connect(mapStateToProps, mapDispatchToProps)(WelcomeScreen);
