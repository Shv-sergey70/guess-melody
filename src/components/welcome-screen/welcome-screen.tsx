import * as React from 'react';
import {ActionCreator} from "../../reducer/game/game";
import {connect} from "react-redux";
import {getTime} from "../../reducer/game/selectors";
import {hasQuestions} from "../../reducer/data/selectors";

type Props = {
  time: number
  attempts: number
  onWelcomeButtonClick: ({}) => void
  isButtonDisabled: boolean
};

const WelcomeScreen: React.FunctionComponent<Props> = ({time, attempts, onWelcomeButtonClick, isButtonDisabled}) => {
  return (
    <section className="welcome">
      <div className="welcome__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/>
      </div>
      <button
        className="welcome__button"
        onClick={onWelcomeButtonClick}
        disabled={isButtonDisabled} >
        <span className="visually-hidden">Начать игру</span>
      </button>
      <h2 className="welcome__rules-title">Правила игры</h2>
      <p className="welcome__text">Правила просты:</p>
      <ul className="welcome__rules-list">
        <li>За {Math.floor(time / 60)} минут нужно ответить на все вопросы.</li>
        <li>Можно допустить {attempts} ошибки.</li>
      </ul>
      <p className="welcome__text">Удачи!</p>
    </section>
  );
};

const mapStateToProps = (state) => ({
  time: getTime(state),
  isButtonDisabled: !hasQuestions(state)
});

const mapDispatchToProps = (dispatch) => ({
  onWelcomeButtonClick: () => dispatch(ActionCreator.incrementStep())
});

export {WelcomeScreen};
export default connect(mapStateToProps, mapDispatchToProps)(WelcomeScreen);
