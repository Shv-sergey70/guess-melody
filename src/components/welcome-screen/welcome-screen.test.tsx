import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {WelcomeScreen} from "./welcome-screen";

configure({adapter: new Adapter()});

describe(`WelcomeScreen correctly works`, () => {
  test(`Start button enabled`, () => {
    const tree = shallow(
        <WelcomeScreen
          time={300}
          attempts={3}
          onWelcomeButtonClick={jest.fn()}
          isButtonDisabled={false}/>
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Start button disabled`, () => {
    const tree = shallow(
        <WelcomeScreen
          time={300}
          attempts={3}
          onWelcomeButtonClick={jest.fn()}
          isButtonDisabled={true}/>
    );

    expect(tree).toMatchSnapshot();
  });
});
