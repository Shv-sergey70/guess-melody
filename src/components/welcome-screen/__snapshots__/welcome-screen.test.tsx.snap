// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen correctly works Start button disabled 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    disabled={true}
    onClick={[MockFunction]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      За 
      5
       минут нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
      3
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;

exports[`WelcomeScreen correctly works Start button enabled 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    disabled={false}
    onClick={[MockFunction]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      За 
      5
       минут нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
      3
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;
